### DISCUSSION TITLE
attributes as private variable

### DISCUSSION THREAD

**Discussion by BartlomiejMajowicz** [laravel/framework#42750] [Q&A] [answered]
https://github.com/laravel/framework/blob/52cea7fb5f7f5e4704518b6a82b21a1e8ecb1cc9/src/Illuminate/Database/Eloquent/Concerns/HasAttributes.php#L38

I faced an issue when column in database was named attributes. Then when I wanted to change row with this column it changed this variable $attributes and created error "Cannot access offset of type string on string" in line 939.

Solution:
When $attributes variable is changed to private this variable is not changed and column with the name is updated.

**Comment by rodrigopedra** [comment]
I could not reproduce it, assigning to a non-public attribute, even when there is protected property named as the same, should trigger the magic method `__set`.

Maybe isn't something else is triggering the error? A trait, a third-party package?

How are you assigning the value to the attributes property?

You can check the execution on PHP from 7.4 to 8.1 (as of this writing) here:

https://3v4l.org/RGIk3

Test code:

~~~php
<?php

class Model {
    protected $attributes = [];

    public function __set($key, $value)
    {
        echo 'called magic method: ', var_export([$key => $value], true), \PHP_EOL;
    }
}

$model = new \Model();
$model->attributes = 'foo';

var_dump($model);
~~~

output

~~~shell
called magic method: array (
  'attributes' => 'foo',
)
object(Model)#1 (1) {
  ["attributes":protected]=>
  array(0) {
  }
}
~~~

**Reply by korkoshko** [reply]
Just try to change the "attributes" property inside the model class (protected properties are available in child classes).

**Reply by rodrigopedra** [reply]
Oh sure... I understood OP wanted to assign a value to an instance, but you are right, if accessed within an instance method it will indeed fail.

**Comment by korkoshko** [comment, marked as answer]
>> When $attributes variable is changed to private this variable is not changed and column with the name is updated.

Many projects use `$this->attributes` on models, this will break everything. 

As a workaround just use `setAttribute` or `setRawAttributes`  methods (depending on the task).

**Reply by BartlomiejMajowicz** [reply]
Okej. Thank You for answer. This can solve the problem.
I already changed to column name to something else.
Anyway, this is a bit confusing.
